pkg/plugins/middlewarewasm.go

```diff
@@ -0,0 +1,80 @@
+package plugins
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"reflect"
+
+	"github.com/http-wasm/http-wasm-host-go/handler"
+	wasm "github.com/http-wasm/http-wasm-host-go/handler/nethttp"
+	"github.com/tetratelabs/wazero"
+	"github.com/traefik/traefik/v3/pkg/logs"
+	"github.com/traefik/traefik/v3/pkg/middlewares"
+)
+
+type wasmMiddlewareBuilder struct {
+	wasmPath string
+}
+
+func newWasmMiddlewareBuilder(goPath string, moduleName string) *wasmMiddlewareBuilder {
+	return &wasmMiddlewareBuilder{
+		wasmPath: filepath.Join(goPath, "src", moduleName, "plugin.wasm"),
```

Comment:
Let's stick with this now but I truly hope we don't end up requiring projects to use `plugin.wasm` convention or them ending up having two releases because `plugin.wasm` isn't meaningful in other ecosystems (e.g. envoy)

Reply:
After a discussion with @juliens we will follow this:

- if there is no explicit filename (or path) in the manifest, use `plugin.wasm`.
- if there is an explicit filename (or path) defined in the manifest, use it.

Reply:
sounds reasonable.